### PR TITLE
Moved and revised the nameof operator article

### PR DESCRIPTION
**Pull request by pkulikov**
Contributes to #10741
Supported by dotnet/samples#1084
@BillWagner this revision makes the article much shorter and more to the point.

### CODE REVIEW

File: docs/visual-basic/getting-started/whats-new.md [hunk 1]
```diff
@@ -180,7 +180,7 @@ For more information, see [Reference Return Values](../programming-guide/languag
 
 ## Visual Basic 14
 
-[Nameof](../../csharp/language-reference/keywords/nameof.md)
+[Nameof](../../csharp/language-reference/operators/nameof.md)
```
Comment (pkulikov) [inline docs/visual-basic/getting-started/whats-new.md]: It seems there is no Visual Basic reference article for the `nameof` operator.

Comment (BillWagner) [reply docs/visual-basic/getting-started/whats-new.md]: /cc @rpetrusha Do you have a better link for this?

File: docs/csharp/language-reference/operators/nameof.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ms.custom: seodec18
+ms.date: 07/12/2019
+f1_keywords:
+  - "nameof_CSharpKeyword"
+  - "nameof"
+ms.assetid: 33601bf3-cc2c-4496-846d-f9679bccf2a7
+---
+# nameof operator (C# reference)
+
+The `nameof` operator obtains the simple name of a variable, type, or member as the string constant:
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/nameof.md]: We should define "simple name". Suggestion:

```suggestion
The `nameof` operator obtains the simple name of a variable, type, or member as the string constant. A [**simple name**](~/_csharplang/spec/expressions#simple-names) is the identifier name. In the case of a generic definition, it is the identifier name followed by any type parameters.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
This looks great @pkulikov 

I had one suggestion for defining "simple name". I'm open to other suggestions there, then this is ready to :shipit:

File: docs/csharp/language-reference/operators/nameof.md (hunk 2 above)
Comment (pkulikov) [reply docs/csharp/language-reference/operators/nameof.md]: @BillWagner I hoped the example would substitute the definition, which is quite difficult to craft.

For example, from the suggestion:
>In the case of a generic definition, it is the identifier name followed by any type parameters.

I would assume that type parameters are included, but the following code
```csharp
Console.WriteLine(nameof(List<int>));
```

produces 
```console
List
```

I've looked at the spec now. It doesn't actually say "simple name", but just a "name". So i suggest to remove the word "simple":
>The `nameof` operator obtains the name of a variable, type, or member as the string constant:

Comment (pkulikov) [reply docs/csharp/language-reference/operators/nameof.md]: @BillWagner I think I was wrong with the word choice in the first version of the article. Could you please take a look at the updated version.

Additionally, I've added the language version, in which the operator was added, and the link from the "What's new..." section.

File: docs/csharp/language-reference/operators/nameof.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  - "nameof"
+ms.assetid: 33601bf3-cc2c-4496-846d-f9679bccf2a7
+---
+# nameof operator (C# reference)
+
+The `nameof` operator obtains the name of a variable, type, or member as the string constant:
+
+[!code-csharp-interactive[nameof operator](~/samples/csharp/language-reference/operators/NameOfOperator.cs#Examples)]
+
+As the preceding example shows, in the case of a type and a namespace, the produced name is not [fully qualified](~/_csharplang/spec/basic-concepts.md#fully-qualified-names).
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/nameof.md]: @BillWagner I'm not technically correct here. The sentence should finish with "...unless a type or a namespace is the member of the global namespace." However I didn't want to over-complicate things. Is the trimmed version fine?

Comment (BillWagner) [reply docs/csharp/language-reference/operators/nameof.md]: I like the way you've worded this. I was looking through the spec this morning to find good language as well. I think a trimmed version is fine, but I'll suggest a bit of weasel words:

```suggestion
As the preceding example shows, in the case of a type and a namespace, the produced name is usually not [fully qualified](~/_csharplang/spec/basic-concepts.md#fully-qualified-names).
```

Comment (pkulikov) [reply docs/csharp/language-reference/operators/nameof.md]: @BillWagner perfect! I've looked for that, but overlooked:) Thanks!

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
I'm happy with how this turned out @pkulikov 

I'm ready to :shipit:, but I'm waiting to see if either @KathleenDollard or @rpetrusha have a better suggestion for the one link on the VB content.

File: docs/visual-basic/getting-started/whats-new.md (hunk 1 above)
Comment (rpetrusha) [reply docs/visual-basic/getting-started/whats-new.md]: There are two choices for the link, both of which should be temporary:
- The mention of `nameof` in https://docs.microsoft.com/en-us/dotnet/visual-basic/getting-started/whats-new#visual-basic-14.
- The design notes, at <https://github.com/dotnet/vblang/blob/189d2c15703b91a9241ce9ade46e6bc24b193cd9/meetings/2014/README.md>.
There's an outstanding issue to add the nameof operator for VB that's assigned to me for this sprint (though it was also assigned to me in the last sprint, and perhaps the one before that). I'll note in the issue that the what's new should be modified to link to the new nameof article when it's ready.

Comment (pkulikov) [reply docs/visual-basic/getting-started/whats-new.md]: @rpetrusha the first suggested link would be a self-reference; the design notes, I'm afraid, would be more confusing than the existing link to the C# `nameof` article (design notes mix C# and VB examples). Thus, I suggest to keep the existing text as it is until the dedicated VB article is created. Please let me know if you prefer the link to the design notes, I'll update the text then.

**Comment by BillWagner** [comment]
Since any link would be temporary, I'll :shipit: now, and we can make a better link choice when the VB article is available.

**Pull request merged**